### PR TITLE
backend/postgres: Various improvments

### PR DESCRIPTION
**Pull request by AlexisMontagne**
### What does this PR do?

This PR packs 3 improvemnt:

* 1- Explicitly handle SSLSNI dsn mode:

lib/pq brings a new sslsni option turns on by default: https://github.com/lib/pq/pull/1088. It breaks our setup with RDS CA.

* 2- Make cancelation error play nice with Go.

if a context is canceled mid request now errors.Is(err, context.Canceled) should return tru

* 3- Accepting multiple CACerts at once.

For CA key  rotation it will be critical 

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [ ] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Properly labeled

### CODE REVIEW

File: backend/postgres/db.go [hunk 1]
```diff
@@ -136,6 +151,10 @@ func wrapErr(err error) error {
 		return err
 	}
 
+	if pqErr.Code == "57014" {
```
Comment (Sypheos) [inline backend/postgres/db.go:154]: no constant in the lib/pq ? :cry:

File: go.mod [hunk 2]
```diff
@@ -5,13 +5,13 @@ go 1.18
 require (
 	github.com/lib/pq v1.4.0
 	github.com/mattn/go-sqlite3 v1.13.1-0.20200416054559-98a44bcf5949
-	github.com/stretchr/testify v1.6.1
+	github.com/stretchr/testify v1.8.4
```
Comment (Sypheos) [inline go.mod:8]: also bump the go version ? (and in the workflow file)

File: backend/postgres/db.go [hunk 3]
```diff
@@ -125,6 +126,20 @@ const (
...
+type queryCanceledError struct {
+	cause *pq.Error
+}
+
+func (qce *queryCanceledError) Error() string {
+	return qce.cause.Error()
+}
+
+func (qce *queryCanceledError) Is(target error) bool {
+	return target == context.Canceled
```
Comment (Sypheos) [inline backend/postgres/db.go:138]: Should we fall back to a the wrapped error if the condition failed ?
```suggestion
	if target == context.Canceled {
	  return true
	}
	
	return errors.Is(qce.cause, target)
```
?

File: backend/postgres/db.go (hunk 1 above)
Comment (AlexisMontagne) [reply backend/postgres/db.go:154]: sadly no

File: backend/postgres/db.go (hunk 3 above)
Comment (AlexisMontagne) [reply backend/postgres/db.go:138]: it is done directly by the std lib in the implementation of `errors.Is`

Comment (Sypheos) [reply backend/postgres/db.go:138]: Won't we need the Unwrap method to be implemented on this type then ?

**Review by xgoffin — APPROVED** [review]
Other than that lgtm

**Review by Sypheos — APPROVED** [review]

**Pull request merged**